src/rest.js

```diff
@@ -57,11 +57,18 @@ function del(config, auth, className, objectId) {
     }
     return Promise.resolve({});
   }).then(() => {
+    if (!auth.isMaster) {
+      return auth.getUserRoles();
```

Comment:
I mean a method on auth like:

function queryACLConstraint() {
 // do something
return {}; // the ACL Constraint for the query
}

This inspired by the way we build the ACL options around the app, this way we centralize that logic.